R/summarize_weighted_production.R

```diff
@@ -262,14 +262,26 @@ add_percent_change <- function(data) {
 
 add_technology_share <- function(data) {
   data %>%
-    group_by(.data$sector_ald, .data$year, .data$scenario, .data$name_ald) %>%
+    group_by(
```

Comment:
Non-blocking: prefer  `[` to `$` as  an accessor (`.data["sector_ald"]`). 

Also, am  I missing a new best practice? why are we expressing this with `.data`?

Reply:
Extracting "Non-blocking: prefer [ to $ as an accessor (.data["sector_ald"])." as a new issue as well, since this would require a rather large style change in the whole package (to keep consistency). 

Regarding `.data`, we would need @maurolepore to chip in, as I am actually not entirely sure. My understanding is that, when using piping, you should explicitly namespace

Reply:
See #335

Reply:
In packages the `.data` pronoun avoids a R CMD NOTE about undefined global variables. You can also define each column you use with `globalVariables(c("column1", "column2"))` or initialize `column1 <- NULL; column2 <- NULL` before the first use, but the `.data` pronoun is the way  the tidyverse recommends. There is some article somewhere -- if curious let me know and I can try find it.